File: GMSFrontend/src/types/offering.ts
export interface Offering {
  id: number;
  term: string;
  courseName: string;
  teacherName: string;
  classLabel?: string | null;
  weekday: number;
  startPeriod: number;
  endPeriod: number;
  startWeek: number;
  endWeek: number;
  location?: string | null;
  capacity: number;
  selectedCount?: number;
  course?: any;
  teacher?: any;
}

export interface TeacherOffering {
  id: number;
  term: string;
  courseName: string;
  classLabel?: string | null;
  timePlace: string;
  studentCount: number;
}

export interface TimetableItem {
  offeringId: number;
  courseName: string;
  teacherName: string;
  weekday: number;
  startPeriod: number;
  endPeriod: number;
  startWeek: number;
  endWeek: number;
  location?: string | null;
}

File: GMSBackend/src/main/java/cn/edu/ctbu/gmsbackend/entity/Offering.java
package cn.edu.ctbu.gmsbackend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Getter
@Setter
@Entity
@com.fasterxml.jackson.annotation.JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Offering {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 32)
    private String term;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id")
    private Course course;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teacher_id")
    private Teacher teacher;

    private String classLabel;

    @Column(nullable = false)
    private Integer weekday;

    @Column(nullable = false)
    private Integer startPeriod;

    @Column(nullable = false)
    private Integer endPeriod;

    @Column(nullable = false)
    private Integer startWeek;

    @Column(nullable = false)
    private Integer endWeek;

    private String location;

    @Column(nullable = false)
    private Integer capacity;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;
}
